code/game/objects/items/weapons/stunbaton.dm

```diff
@@ -83,6 +83,22 @@
 /obj/item/melee/baton/get_cell()
 	return cell
 
+/obj/item/melee/baton/mob_can_equip(mob/user, slot, disable_warning = 1)
+	if(turned_on && slot == slot_belt)
+		to_chat(user, "<span class='warning'>You can't equip [src] while it's active!</span>")
+		return FALSE
+	if(turned_on && slot == slot_s_store)
+		to_chat(user, "<span class='warning'>You can't equip [src] while it's active!</span>")
+		return FALSE
+	return ..()
```

Comment:
```suggestion
	return ..(user, slot, disable_warning == TRUE) // call parent but disable warning
```
you should just be able to do this instead of changing the default param